openbas-api/src/main/java/io/openbas/scheduler/jobs/InjectsExecutionJob.java

```diff
@@ -225,6 +235,88 @@ private void executeInject(ExecutableInject executableInject) {
         }
     }
 
+    /**
+     * Get error messages if pre execution conditions are not met
+     * @param exerciseId the id of the exercise
+     * @param inject the inject to check
+     * @return an optional of list of error message
+     */
+    private Optional<List<String>> getErrorMessagesPreExecution(String exerciseId, Inject inject) {
+        List<InjectDependency> injectDependencies = injectDependenciesRepository.findParents(List.of(inject.getId()));
+        if (!injectDependencies.isEmpty()) {
+            List<Inject> parents = StreamSupport.stream(injectRepository.findAllById(injectDependencies.stream()
+                    .map(injectDependency -> injectDependency.getCompositeId().getInjectParentId().getId()).toList()).spliterator(), false)
+                    .collect(Collectors.toList());
+
+            Map<String, Boolean> mapCondition = getStringBooleanMap(parents, exerciseId, injectDependencies);
+
+            List<String> results = null;
+
+            for (InjectDependency injectDependency : injectDependencies) {
+                String expressionToEvaluate = injectDependency.getCondition();
+                List<String> conditions = Arrays.stream(injectDependency.getCondition()
```

Comment:
Maybe I'm overcomplicating things for nothing here but I chose to use a string to avoid the complexity of having to deal with complex boolean expressions with parenthesis negations. I guess we could use a tree structure but I'm unsure it would be easier to undestand and manage. Did you had something else in mind ?

Reply:
I have a tree structure in mind like for the filter mechanism.
My concern is that handling a string that has a complex structure can be error-prone.
If the string is modified for some reason, this could be a source of error.
Moreover on front side instead of having simple methods like getOperator, getCondition, we need to perform parsing on particular keys.

We can have discussion on this point

Reply:
Since we're doing this, we might as well do https://github.com/OpenBAS-Platform/openbas/pull/1551/files#r1801381652 since both impact the same thing. I'm on it